src/devices/Nmea0183/tests/Ais/AisManagerTest.cs

```diff
@@ -0,0 +1,415 @@
+
+            foreach (var ship in _manager.GetTargets())
+            {
+                if (!ship.Position.ContainsValidPosition())
+                {
+                    continue;
+                }
+
+                var distance = ownShip.DistanceTo(ship);
+                distance.ShouldBeGreaterThan(Length.FromMeters(50));
+                distance.ShouldBeLessThan(Length.FromNauticalMiles(30));
+
+                var age = ship.Age(latestPacketDate);
+                if (age.Duration() > TimeSpan.FromMinutes(5))
+                {
+                    continue;
+                }
+
+                var relativePosition = ownShip.RelativePositionTo(ship, latestPacketDate, new TrackEstimationParameters());
+                Assert.NotNull(relativePosition);
+                Assert.True(relativePosition!.From == ownShip);
+                Assert.True(relativePosition.To == ship);
+                // Some error is acceptable, since "distance" is not corrected for the time between the now and the last position
+                Assert.True(Math.Abs(distance.Meters - relativePosition.Distance.Meters) < 100);
+            }
+
+            var ship1 = _manager.GetTarget(211810280) as Ship;
+            Assert.NotNull(ship1);
+            var relativePos1 = ownShip.RelativePositionTo(ship1!, latestPacketDate, new TrackEstimationParameters());
+            Assert.NotNull(relativePos1);
+            Assert.True(relativePos1!.TimeOfClosestPointOfApproach.HasValue);
+            Assert.True(relativePos1.ClosestPointOfApproach.HasValue);
+
+            Assert.True(relativePos1.ClosestPointOfApproach < relativePos1.Distance);
+            Assert.True(relativePos1.TimeToClosestPointOfApproach(latestPacketDate) < TimeSpan.Zero);
+
+            // this ship is directly moving towards us at the end of the test sequence (if we didn't move, and ignoring that
+            // in reality, the test data set is from a river with many bends and even drawbridges)
+            ship1 = _manager.GetTarget(305966000) as Ship;
+            Assert.NotNull(ship1);
+            ownShip.SpeedOverGround = Speed.Zero; // Easier to get a close passage if we don't move here
+            relativePos1 = ownShip.RelativePositionTo(ship1!, latestPacketDate, new TrackEstimationParameters());
+            Assert.NotNull(relativePos1);
+            Assert.True(relativePos1!.TimeOfClosestPointOfApproach.HasValue);
+            Assert.True(relativePos1.ClosestPointOfApproach.HasValue);
+
+            Assert.True(relativePos1.ClosestPointOfApproach < relativePos1.Distance);
+            Assert.Equal(1790, relativePos1.TimeToClosestPointOfApproach(latestPacketDate)!.Value.TotalSeconds, 0);
+        }
+
+        [Fact]
+        public void CheckSafetyPermanently()
+        {
+            // This does a safety check all the time. Very expensive...
+            using NmeaLogDataReader reader = new NmeaLogDataReader("Reader", "../../../Nmea-2021-08-25-16-25.txt");
```

Comment:
might be worth creating utility `GetDataLogger` so you can change this easier everywhere if needed